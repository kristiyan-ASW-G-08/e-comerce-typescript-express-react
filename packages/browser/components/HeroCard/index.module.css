.card1 {
  background-color: #475569;
  background-image: url("data:image/svg+xml,%3Csvg width='6' height='6' viewBox='0 0 6 6' xmlns='http://www.w3.org/2000/svg'%3E%3Cg fill='%23f1f5f9' fill-opacity='0.4' fill-rule='evenodd'%3E%3Cpath d='M5 0h1L0 6V5zM6 5v1H5z'/%3E%3C/g%3E%3C/svg%3E");
}
.card2 {
  background-color: #ea580c;
  background-image: url("data:image/svg+xml,%3Csvg width='6' height='6' viewBox='0 0 6 6' xmlns='http://www.w3.org/2000/svg'%3E%3Cg fill='%23ffedd5' fill-opacity='0.4' fill-rule='evenodd'%3E%3Cpath d='M5 0h1L0 6V5zM6 5v1H5z'/%3E%3C/g%3E%3C/svg%3E");
}
.card3 {
  background-color: #9333ea;
  background-image: url("data:image/svg+xml,%3Csvg width='6' height='6' viewBox='0 0 6 6' xmlns='http://www.w3.org/2000/svg'%3E%3Cg fill='%23f3e8ff' fill-opacity='0.4' fill-rule='evenodd'%3E%3Cpath d='M5 0h1L0 6V5zM6 5v1H5z'/%3E%3C/g%3E%3C/svg%3E");
}
.card4 {
}
