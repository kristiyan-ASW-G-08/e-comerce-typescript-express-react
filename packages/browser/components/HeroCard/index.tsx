import { FC } from 'react';
import Image from 'next/image';
import styles from './index.module.css';
interface HeroCardProps {
  cardNum: string;
  content: string;
  heading: string;
  imageSrc: any;
}
const HeroCard: FC<HeroCardProps> = ({
  cardNum,
  imageSrc,
  content,
  heading,
}) => (
  <div
    className={`${styles[cardNum]} animate-fade h-44 mb-4 rounded flex flex-row mx-4 sm:mx-4 w-full lg:w-1/3 p-3`}
  >
    <div className=" flex flex-col m-auto">
      <h1 className="text-2xl  text-neutral-50">{heading}</h1>
      <p className="text-base  text-neutral-50">{content}</p>
    </div>
    <div className="h-40 w-40 relative m-auto">
      <Image layout="fill" alt="" src={imageSrc} />
    </div>
  </div>
);

export default HeroCard;
