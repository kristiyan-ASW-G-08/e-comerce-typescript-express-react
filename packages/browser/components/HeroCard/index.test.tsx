import React from 'react';
import { render } from '@testing-library/react';
import HeroCard from '.';
import fileMock from '../../mocks/fileMock';
describe('HeroCard', () => {
  afterEach(() => jest.resetAllMocks());
  afterAll(() => jest.restoreAllMocks());
  it('render ', async () => {
    // expect.assertions(2);
    const heading = 'Browse Our Promotions';
    const imageSrc = 'some.svg';
    const cardNum = 'card1';
    const content = 'Wireless Connection With TV, Computer, Laptop..';
    const { container } = render(
      <HeroCard
        heading={heading}
        imageSrc={fileMock}
        cardNum={cardNum}
        content={content}
      />,
      {},
    );

    expect(container).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
