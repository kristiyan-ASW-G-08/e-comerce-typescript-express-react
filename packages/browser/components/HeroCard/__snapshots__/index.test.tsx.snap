// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroCard render  1`] = `
<div>
  <div
    class="undefined animate-fade h-44 mb-4 rounded flex flex-row mx-4 sm:mx-4 w-full lg:w-1/3 p-3"
  >
    <div
      class=" flex flex-col m-auto"
    >
      <h1
        class="text-2xl  text-neutral-50"
      >
        Browse Our Promotions
      </h1>
      <p
        class="text-base  text-neutral-50"
      >
        Wireless Connection With TV, Computer, Laptop..
      </p>
    </div>
    <div
      class="h-40 w-40 relative m-auto"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt=""
          data-nimg="fill"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
  </div>
</div>
`;
